JPS_VIRTUALSENSOR/JurongIslandInputAgent/src/main/java/com/cmclinnovations/jurongisland/QueryClient.java

```diff
@@ -0,0 +1,227 @@
+
+    }
+
+    public void updateEmissionsData(JSONArray pollutantSourceData) {
+
+        ModifyQuery modify = Queries.MODIFY().prefix(P_DISP,P_OWL,P_OM);
+
+        for (int i = 0; i < pollutantSourceData.length(); i++) {
+
+            String ocgmlIRI = pollutantSourceData.getJSONObject(i).getString("IRI");
+            Double emission = pollutantSourceData.getJSONObject(i).getDouble("emission");
+            RdfLiteral.NumericLiteral emissionValue = Rdf.literalOf(emission);
+            double density = Double.parseDouble(EnvConfig.DENSITY);
+            RdfLiteral.NumericLiteral densityValue = Rdf.literalOf(density);
+            double temp = Double.parseDouble(EnvConfig.TEMPERATURE);
+            RdfLiteral.NumericLiteral tempValue = Rdf.literalOf(temp);
+
+            String pollutantSourceIRI = PREFIX + "staticpointsource/" + UUID.randomUUID();
+            String emissionIRI = PREFIX + "co2/" + UUID.randomUUID();
+            String densityIRI = PREFIX + "density/" + UUID.randomUUID();
+            String massFlowIRI = PREFIX + "massflow/" + UUID.randomUUID();
+            String temperatureIRI = PREFIX + "temperature/" + UUID.randomUUID();
+
+            modify.insert(iri(pollutantSourceIRI).isA(STATIC_POINT_SOURCE).andHas(EMITS,iri(emissionIRI)));
+            modify.insert(iri(emissionIRI).isA(CO2).andHas(HAS_QUANTITY,iri(densityIRI))
+            .andHas(HAS_QUANTITY,iri(massFlowIRI)).andHas(HAS_QUANTITY,iri(temperatureIRI)));
+
+            modify.insert(iri(emissionIRI).has(HAS_OCGML_OBJECT,iri(ocgmlIRI)));
+            modify.insert(iri(ocgmlIRI).isA(OWL_THING));
+
+            // Triples for density, mass flow rate and temperature
+            modify.insert(iri(densityIRI).isA(DENSITY).andHas(HAS_NUMERICALVALUE,densityValue).andHas(HAS_UNIT,DENSITY_UNIT));
+            modify.insert(iri(massFlowIRI).isA(MASS_FLOW).andHas(HAS_NUMERICALVALUE,emissionValue).andHas(HAS_UNIT,MASS_FLOW_UNIT));
+            modify.insert(iri(temperatureIRI).isA(TEMPERATURE).andHas(HAS_NUMERICALVALUE,tempValue).andHas(HAS_UNIT,TEMPERATURE_UNIT));
+
+        }
+
+        updateStoreClient.executeUpdate(modify.getQueryString());
+
+    }
+
+    public void updatePirmasensEmissions() {
+        List<String> ocgmlIRIs = Arrays.asList(
+            "http://www.theworldavatar.com:83/citieskg/namespace/pirmasensEPSG32633/sparql/cityobject/UUID_LOD2_Pirmasens_4f8d0f1a-3b21-40d4-8b90-89723e31a7ca/",
+            "http://www.theworldavatar.com:83/citieskg/namespace/pirmasensEPSG32633/sparql/cityobject/UUID_LOD2_Pirmasens_c38d038b-a677-4e0c-95d9-f02c09cf991c/");
+            List<Double> no2Emissions = Arrays.asList(100.0,100.0);
+            List<Double> pm25Emissions = Arrays.asList(100.0,100.0);
+            List<Double> pm10Emissions = Arrays.asList(100.0,100.0);
+
+            List<Double> no2Densities = Arrays.asList(1.0,1.0);
+            List<Double> pm25Densities = Arrays.asList(1.0,1.0);
+            List<Double> pm10Densities = Arrays.asList(1.0,1.0);
```

Comment:
In the future, try to use object oriented programming instead of creating 7 lists